prow/plugins/plugins.go

```diff
@@ -131,14 +131,32 @@ type PluginAgent struct {
 // target for plugin Configuration
 type Configuration struct {
 	// Repo (eg "k/k") -> list of handler names.
-	Plugins         map[string][]string `json:"plugins,omitempty"`
-	Triggers        []Trigger           `json:"triggers,omitempty"`
-	Heart           Heart               `json:"heart,omitempty"`
-	MilestoneStatus MilestoneStatus     `json:"milestonestatus,omitempty"`
-	Slack           Slack               `json:"slack,omitempty"`
-	// ConfigUpdater holds config for the config-updater plugin.
-	ConfigUpdater ConfigUpdater `json:"config_updater,omitempty"`
-	Blockades     []Blockade    `json:"blockades,omitempty"`
+	Plugins map[string][]string `json:"plugins,omitempty"`
+
+	// ExternalPlugins is a list of external plugin names mapped
+	// to Kubernetes Service names.
+	ExternalPlugins []ExternalPlugin `json:"external_plugins"`
+
+	// Prow build-in plugins
+	Triggers        []Trigger       `json:"triggers,omitempty"`
+	Heart           Heart           `json:"heart,omitempty"`
+	MilestoneStatus MilestoneStatus `json:"milestonestatus,omitempty"`
+	Slack           Slack           `json:"slack,omitempty"`
+	ConfigUpdater   ConfigUpdater   `json:"config_updater,omitempty"`
+	Blockades       []Blockade      `json:"blockades,omitempty"`
+}
+
+// ExternalPlugin holds configuration for registering an external
+// plugin in prow.
+type ExternalPlugin struct {
```

Comment:
Without a token here anyone who can resolve the service name can execute code with potentially privileged tokens. Do we care enough to add HMAC tokens here?

Reply:
I am reusing the same HMAC token that is used by `hook`. Notice that the payload is forwarded to the external plugins as is. We could use separate tokens in the future but that would require unmarshaling the payload, encrypting with the new token, marshaling, and then sending it. I prefer to avoid the extra steps for now.